nats-exporter/Dockerfile

```diff
@@ -14,23 +14,23 @@
 # limitations under the License.
 
 # Golang binary building stage
-ARG GO_VERSION=1.21
+ARG GO_VERSION=1.23
```

Comment:
the underlying golang version for nats exporter for 1.15 is 1.21 https://github.com/nats-io/prometheus-nats-exporter/blob/v0.15.0/go.mod#L3

Reply:
It builds fine on 1.23 though. Is 1.23 going to cause problems?

Reply:
nats-exporter 0.16.0 uses golang 1.23, so i'm going to bump it to that. https://github.com/nats-io/prometheus-nats-exporter/releases/tag/v0.16.0